examples/jetstream/push-consumer/java/Main.java

```diff
@@ -0,0 +1,198 @@
+package example;
+
+import io.nats.client.*;
+import io.nats.client.api.*;
+import io.nats.client.impl.NatsJetStreamMetaData;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+
+public class Main {
+    public static void main(String[] args) {
+        String natsURL = System.getenv("NATS_URL");
+        if (natsURL == null) {
+            natsURL = "nats://127.0.0.1:4222";
+        }
+
+        // Initialize a connection to the server. The connection is AutoCloseable
+        // on exit.
+        try (Connection nc = Nats.connect(natsURL)) {
+
+            // Access `JetStream` and `JetStreamManagement` which provide methods to create
+            // streams and consumers as well as convenience methods for publishing
+            // to streams and consuming messages from the streams.
+            JetStream js = nc.jetStream();
+            JetStreamManagement jsm = nc.jetStreamManagement();
+
+            // Declare a simple [limits-based stream][1] and populate the stream
+            // with a few messages.
+            // [1]: /examples/jetstream/limits-stream/java/
+            String streamName = "EVENTS";
+            StreamConfiguration config = StreamConfiguration.builder()
+                    .name(streamName)
+                    .subjects("events.>")
+                    .build();
+
+            StreamInfo stream = jsm.addStream(config);
+
+            js.publish("events.1", null);
+            js.publish("events.2", null);
+            js.publish("events.3", null);
+
+            // ### Ephemeral
+            // The JetStream context provides a simple way to create an ephemeral
+            // push consumer, simply provide a subject that overlaps with the
+            // bound subjects on the stream and this helper method will do the
+            // stream look-up automatically and create the consumer.
+            System.out.println("# Ephemeral");
+            Subscription sub = js.subscribe("events.>");
+
+            // An ephemeral consumer has a name generated on the server-side.
```

Comment:
Maybe say that the consumer name is generated if not provided.

Reply:
Done